pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -32,7 +32,7 @@ def on_trainer_init(
         # it is important that this is the last callback to run
         # pass through the required args to figure out defaults
         checkpoint_callback = self.init_default_checkpoint_callback(checkpoint_callback)
-        if checkpoint_callback:
+        if checkpoint_callback and not resume_from_checkpoint:
```

Comment:
why isn't this logic pushed into `init_default_checkpoint_callback` ? for example, it could loop through all of the trainer's set callbacks and return None/False if any is of type `ModelCheckpoint`

Reply:
yes!
It's exactly what I am trying to do here: #3990
I want to enable passing the checkpoint callback to the callbacks list directly. Then ran into this problem here.

Reply:
ah so the default checkpoint is init-ed, then we restore from checkpoint, and that leads to the duplication? I think this check could still happen inside `init_default_checkpoint_callback` so the logic is all in one place for the returned `checkpoint_callback` here

Reply:
- why isn't the restore the first thing to happen? 
- what happens if someone resumes from checkpoint but they don't want to save a checkpoint? 
- what happens if someone resumes from checkpoint but they want to use a different checkpoint callback than the one that was in the original checkpoint?